lib/vault/client.rb

```diff
@@ -132,7 +135,7 @@ def request(verb, path, data = {}, headers = {})
       # Add the Vault token header - users could still override this on a
       # per-request basis
       if !token.nil?
-        request.add_field(TOKEN_HEADER, token)
+        headers[TOKEN_HEADER] ||= token
```

Comment:
Experiments with overriding this header field via the header parameters hash were not working as expected (ruby 2.3.0).  This change fixed that allowing the user to override the value, as per the expectation, as well as consolidating the generation of headers on the request object.